File: Dart/web/Model/TStep.dart
import 'Database.dart';

class TStep {
  int Id;
  String Text;
  int ParentId;
  bool Done;

  TStep(
      {int id = -1,
      String text = '',
      int parentId = UndividedTListId,
      bool done = false}) {
    Id = id;
    Text = text;
    ParentId = parentId;
    Done = done;
  }

  void Update(Function callback) {
    Database.Current.UpdateTStep(this, callback: callback);
  }

  void Remove(Function callback) {
    Database.Current.RemoveTStep(Id, callback: callback);
  }
}

File: TypeScript/TSTodo/wwwroot/Model/TStep.ts
class TStep {
    public Id: number;
    public Text: string;
    public ParentId: number;
    public Done: boolean;

    public constructor(id: number = -1, text: string = "", parentId: number = UndividedTListId,
        done: boolean = false) {
        this.Id = id;
        this.Text = text;
        this.ParentId = parentId;
        this.Done = done;
    }

    public Update(callback): void {
        Database.Current().UpdateTStep(this, callback);
    }

    public Remove(callback): void {
        Database.Current().RemoveTStep(this.Id, callback);
    }
}